extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js

```diff
@@ -364,8 +370,17 @@ export class LightboxManager {
     return null;
   }
 
+  getVideoTimestamp_(element) {
+    if (VIDEO_TAGS[element.tagName]) {
+      return element.getImpl().then(videoPlayer => {
+        return videoPlayer.getDuration();
+      });
+    } else {
+      return null;
```

Comment:
When methods return a Promise, I try to avoid returning `null` for an empty response [to avoid Zalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony), or in other words mixing synchronous and asynchronous behavior.

Return an empty `Promise.resolve()` instead and do the empty check in the consumer's `.then()` 

e.g. 

```js
this.getVideoTimestamp_().then(ts => {
  if (!ts) {
    return;
  }
  /* ... */
});